File: php/GitHub/Models/PullRequestSimple__links.php
<?php

namespace Baywet\Githubtododemo\Github\Models;

use Microsoft\Kiota\Abstractions\Serialization\AdditionalDataHolder;
use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

class PullRequestSimple__links implements AdditionalDataHolder, Parsable 
{
    /**
     * @var array<string, mixed>|null $additionalData Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    */
    private ?array $additionalData = null;
    
    /**
     * @var Link|null $comments Hypermedia Link
    */
    private ?Link $comments = null;
    
    /**
     * @var Link|null $commits Hypermedia Link
    */
    private ?Link $commits = null;
    
    /**
     * @var Link|null $escapedSelf Hypermedia Link
    */
    private ?Link $escapedSelf = null;
    
    /**
     * @var Link|null $html Hypermedia Link
    */
    private ?Link $html = null;
    
    /**
     * @var Link|null $issue Hypermedia Link
    */
    private ?Link $issue = null;
    
    /**
     * @var Link|null $review_comment Hypermedia Link
    */
    private ?Link $review_comment = null;
    
    /**
     * @var Link|null $review_comments Hypermedia Link
    */
    private ?Link $review_comments = null;
    
    /**
     * @var Link|null $statuses Hypermedia Link
    */
    private ?Link $statuses = null;
    
    /**
     * Instantiates a new pullRequestSimple__links and sets the default values.
    */
    public function __construct() {
        $this->setAdditionalData([]);
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return PullRequestSimple__links
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): PullRequestSimple__links {
        return new PullRequestSimple__links();
    }

    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return array<string, mixed>|null
    */
    public function getAdditionalData(): ?array {
        return $this->additionalData;
    }

    /**
     * Gets the comments property value. Hypermedia Link
     * @return Link|null
    */
    public function getComments(): ?Link {
        return $this->comments;
    }

    /**
     * Gets the commits property value. Hypermedia Link
     * @return Link|null
    */
    public function getCommits(): ?Link {
        return $this->commits;
    }

    /**
     * Gets the self property value. Hypermedia Link
     * @return Link|null
    */
    public function getEscapedSelf(): ?Link {
        return $this->escapedSelf;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return  [
            'comments' => fn(ParseNode $n) => $o->setComments($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'commits' => fn(ParseNode $n) => $o->setCommits($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'self' => fn(ParseNode $n) => $o->setEscapedSelf($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'html' => fn(ParseNode $n) => $o->setHtml($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'issue' => fn(ParseNode $n) => $o->setIssue($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'review_comment' => fn(ParseNode $n) => $o->setReviewComment($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'review_comments' => fn(ParseNode $n) => $o->setReviewComments($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
            'statuses' => fn(ParseNode $n) => $o->setStatuses($n->getObjectValue([Link::class, 'createFromDiscriminatorValue'])),
        ];
    }

    /**
     * Gets the html property value. Hypermedia Link
     * @return Link|null
    */
    public function getHtml(): ?Link {
        return $this->html;
    }

    /**
     * Gets the issue property value. Hypermedia Link
     * @return Link|null
    */
    public function getIssue(): ?Link {
        return $this->issue;
    }

    /**
     * Gets the review_comment property value. Hypermedia Link
     * @return Link|null
    */
    public function getReviewComment(): ?Link {
        return $this->review_comment;
    }

    /**
     * Gets the review_comments property value. Hypermedia Link
     * @return Link|null
    */
    public function getReviewComments(): ?Link {
        return $this->review_comments;
    }

    /**
     * Gets the statuses property value. Hypermedia Link
     * @return Link|null
    */
    public function getStatuses(): ?Link {
        return $this->statuses;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        $writer->writeObjectValue('comments', $this->getComments());
        $writer->writeObjectValue('commits', $this->getCommits());
        $writer->writeObjectValue('self', $this->getEscapedSelf());
        $writer->writeObjectValue('html', $this->getHtml());
        $writer->writeObjectValue('issue', $this->getIssue());
        $writer->writeObjectValue('review_comment', $this->getReviewComment());
        $writer->writeObjectValue('review_comments', $this->getReviewComments());
        $writer->writeObjectValue('statuses', $this->getStatuses());
        $writer->writeAdditionalData($this->getAdditionalData());
    }

    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param array<string,mixed> $value Value to set for the AdditionalData property.
    */
    public function setAdditionalData(?array $value): void {
        $this->additionalData = $value;
    }

    /**
     * Sets the comments property value. Hypermedia Link
     * @param Link|null $value Value to set for the comments property.
    */
    public function setComments(?Link $value): void {
        $this->comments = $value;
    }

    /**
     * Sets the commits property value. Hypermedia Link
     * @param Link|null $value Value to set for the commits property.
    */
    public function setCommits(?Link $value): void {
        $this->commits = $value;
    }

    /**
     * Sets the self property value. Hypermedia Link
     * @param Link|null $value Value to set for the self property.
    */
    public function setEscapedSelf(?Link $value): void {
        $this->escapedSelf = $value;
    }

    /**
     * Sets the html property value. Hypermedia Link
     * @param Link|null $value Value to set for the html property.
    */
    public function setHtml(?Link $value): void {
        $this->html = $value;
    }

    /**
     * Sets the issue property value. Hypermedia Link
     * @param Link|null $value Value to set for the issue property.
    */
    public function setIssue(?Link $value): void {
        $this->issue = $value;
    }

    /**
     * Sets the review_comment property value. Hypermedia Link
     * @param Link|null $value Value to set for the review_comment property.
    */
    public function setReviewComment(?Link $value): void {
        $this->review_comment = $value;
    }

    /**
     * Sets the review_comments property value. Hypermedia Link
     * @param Link|null $value Value to set for the review_comments property.
    */
    public function setReviewComments(?Link $value): void {
        $this->review_comments = $value;
    }

    /**
     * Sets the statuses property value. Hypermedia Link
     * @param Link|null $value Value to set for the statuses property.
    */
    public function setStatuses(?Link $value): void {
        $this->statuses = $value;
    }

}

File: java/app/src/main/java/githubtododemo/githubclient/models/PullRequestSimpleLinks.java
package githubtododemo.githubclient.models;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
@jakarta.annotation.Generated("com.microsoft.kiota")
public class PullRequestSimpleLinks implements Parsable {
    /**
     * Hypermedia Link
     */
    private Link comments;
    /**
     * Hypermedia Link
     */
    private Link commits;
    /**
     * Hypermedia Link
     */
    private Link html;
    /**
     * Hypermedia Link
     */
    private Link issue;
    /**
     * Hypermedia Link
     */
    private Link reviewComment;
    /**
     * Hypermedia Link
     */
    private Link reviewComments;
    /**
     * Hypermedia Link
     */
    private Link self;
    /**
     * Hypermedia Link
     */
    private Link statuses;
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a PullRequestSimpleLinks
     */
    @jakarta.annotation.Nonnull
    public static PullRequestSimpleLinks createFromDiscriminatorValue(@jakarta.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new PullRequestSimpleLinks();
    }
    /**
     * Gets the comments property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getComments() {
        return this.comments;
    }
    /**
     * Gets the commits property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getCommits() {
        return this.commits;
    }
    /**
     * The deserialization information for the current model
     * @return a Map<String, java.util.function.Consumer<ParseNode>>
     */
    @jakarta.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(8);
        deserializerMap.put("comments", (n) -> { this.setComments(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("commits", (n) -> { this.setCommits(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("html", (n) -> { this.setHtml(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("issue", (n) -> { this.setIssue(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("review_comment", (n) -> { this.setReviewComment(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("review_comments", (n) -> { this.setReviewComments(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("self", (n) -> { this.setSelf(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("statuses", (n) -> { this.setStatuses(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        return deserializerMap;
    }
    /**
     * Gets the html property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getHtml() {
        return this.html;
    }
    /**
     * Gets the issue property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getIssue() {
        return this.issue;
    }
    /**
     * Gets the review_comment property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getReviewComment() {
        return this.reviewComment;
    }
    /**
     * Gets the review_comments property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getReviewComments() {
        return this.reviewComments;
    }
    /**
     * Gets the self property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getSelf() {
        return this.self;
    }
    /**
     * Gets the statuses property value. Hypermedia Link
     * @return a Link
     */
    @jakarta.annotation.Nullable
    public Link getStatuses() {
        return this.statuses;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public void serialize(@jakarta.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeObjectValue("comments", this.getComments());
        writer.writeObjectValue("commits", this.getCommits());
        writer.writeObjectValue("html", this.getHtml());
        writer.writeObjectValue("issue", this.getIssue());
        writer.writeObjectValue("review_comment", this.getReviewComment());
        writer.writeObjectValue("review_comments", this.getReviewComments());
        writer.writeObjectValue("self", this.getSelf());
        writer.writeObjectValue("statuses", this.getStatuses());
    }
    /**
     * Sets the comments property value. Hypermedia Link
     * @param value Value to set for the comments property.
     */
    public void setComments(@jakarta.annotation.Nullable final Link value) {
        this.comments = value;
    }
    /**
     * Sets the commits property value. Hypermedia Link
     * @param value Value to set for the commits property.
     */
    public void setCommits(@jakarta.annotation.Nullable final Link value) {
        this.commits = value;
    }
    /**
     * Sets the html property value. Hypermedia Link
     * @param value Value to set for the html property.
     */
    public void setHtml(@jakarta.annotation.Nullable final Link value) {
        this.html = value;
    }
    /**
     * Sets the issue property value. Hypermedia Link
     * @param value Value to set for the issue property.
     */
    public void setIssue(@jakarta.annotation.Nullable final Link value) {
        this.issue = value;
    }
    /**
     * Sets the review_comment property value. Hypermedia Link
     * @param value Value to set for the review_comment property.
     */
    public void setReviewComment(@jakarta.annotation.Nullable final Link value) {
        this.reviewComment = value;
    }
    /**
     * Sets the review_comments property value. Hypermedia Link
     * @param value Value to set for the review_comments property.
     */
    public void setReviewComments(@jakarta.annotation.Nullable final Link value) {
        this.reviewComments = value;
    }
    /**
     * Sets the self property value. Hypermedia Link
     * @param value Value to set for the self property.
     */
    public void setSelf(@jakarta.annotation.Nullable final Link value) {
        this.self = value;
    }
    /**
     * Sets the statuses property value. Hypermedia Link
     * @param value Value to set for the statuses property.
     */
    public void setStatuses(@jakarta.annotation.Nullable final Link value) {
        this.statuses = value;
    }
}
